### PR TITLE
Cooler tests

### PR DESCRIPTION
**Pull request by kvish**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kvish__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (kavishw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/StopCommandProcessorTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        public async void TestCommandRetryLater()
+        {
+            var history = new CommandHistory("StopTelemetry");
+            var command = new DeserializableCommand(history, "LockToken");
+            
+            var r = await _stopCommandProcessor.HandleCommandAsync(command);
+            Assert.Equal(r, CommandProcessingResult.RetryLater);
+        }
+    }
+}
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/StopCommandProcessorTests.cs]: test for Success ?

File: DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/StartCommandProcessorTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        public async void TestCommandRetryLater()
+        {
+            var history = new CommandHistory("StartTelemetry");
+            var command = new DeserializableCommand(history, "LockToken");
+
+            var r = await _startCommandProcessor.HandleCommandAsync(command);
+            Assert.Equal(r, CommandProcessingResult.RetryLater);
+        }
+    }
+}
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/StartCommandProcessorTests.cs]: test for Success ?

File: DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/DiagnosticTelemetryCommandProcessorTests.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        [Fact]
+        public async void RetryLaterCommandTests()
+        {
+            var history = new CommandHistory("DiagnosticTelemetry");
+            var command = new DeserializableCommand(history, "LockToken");
+            history.Parameters = new ExpandoObject();
+            history.Parameters.Active = true;
+            //null pararameters
+            var r = await _diagnosticTelemetryCommandProcessor.HandleCommandAsync(command);
+            Assert.Equal(r, CommandProcessingResult.RetryLater);
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/DiagnosticTelemetryCommandProcessorTests.cs]: Success ?

File: DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/ChangeSetPointTempCommandProcessorTests.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        public async void CannotParseAsDoubleCommandTests()
+        {
+            var history = new CommandHistory("ChangeSetPointTemp");
+            var command = new DeserializableCommand(history, "LockToken");
+            history.Parameters = new ExpandoObject();
+            history.Parameters.SetPointTemp = "ThisIsNotADouble";
+
+            var r = await _changeSetPointTempCommandProcessor.HandleCommandAsync(command);
+            Assert.Equal(r, CommandProcessingResult.CannotComplete);
+        }
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/ChangeSetPointTempCommandProcessorTests.cs]: Success ?

**Pull request merged**